pgsqltoolsservice/driver/types/driver.py

```diff
@@ -0,0 +1,100 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from typing import List, Mapping, Tuple
+from abc import ABC, abstractmethod
+from pgsqltoolsservice.parsers.owner_uri_parser import get_attribute_value
+
+class ServerConnection(ABC):
+    """Abstract base class that outlines methods and properties that connections must implement"""
+
+    ###################### PROPERTIES ##################################
+    @property
+    @abstractmethod
+    def connection(self):
```

Comment:
What would be type of this ? If we dont define a public interface for this we might have specific logic w.r.t providers

Reply:
I understand the issue here, as the type would depend on what the underlying driver is. I think in this case, in the interest of abstraction and encapsulation, we should not expose the underlying connection through a public property, so I removed this method for now.